File: app/src/main/java/com/example/learn/testme/ViewHolder/CategoryViewHolder.java
package com.example.learn.testme.ViewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.learn.testme.Interface.ItemClickLIstener;
import com.example.learn.testme.R;

public class CategoryViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{

    public TextView category_name;
    public ImageView category_image;
    private ItemClickLIstener itemClickListener;



    public CategoryViewHolder(View itemView) {
        super(itemView);
        category_image = (ImageView)itemView.findViewById(R.id.category_image);
        category_name = (TextView)itemView.findViewById(R.id.category_name);
        itemView.setOnClickListener(this);
    }

    public void setItemClickListener(ItemClickLIstener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        itemClickListener.onClick(view ,getAdapterPosition(),false);


    }

}

File: app/src/main/java/com/example/learn/testme/ViewHolder/CategoryViewHolder.kt
package com.example.learn.testme.ViewHolder

import androidx.recyclerview.widget.RecyclerView.ViewHolder
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.example.learn.testme.Interface.ItemClickLIstener
import com.example.learn.testme.R

class CategoryViewHolder(itemView: View) : ViewHolder(itemView), View.OnClickListener {
    var category_name: TextView
    var category_image: ImageView
    private var itemClickListener: ItemClickLIstener? = null
    fun setItemClickListener(itemClickListener: ItemClickLIstener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(view: View) {
        itemClickListener!!.onClick(view, adapterPosition, false)
    }

    init {
        category_image = itemView.findViewById<View>(R.id.category_image) as ImageView
        category_name = itemView.findViewById<View>(R.id.category_name) as TextView
        itemView.setOnClickListener(this)
    }
}